gui/default/syncthing/core/syncthingController.js

```diff
@@ -1729,11 +1747,24 @@ angular.module('syncthing.core')
             $('#needed').modal().on('hidden.bs.modal', function () {
                 $scope.neededFolder = undefined;
                 $scope.needed = undefined;
-                $scope.neededTotal = 0;
                 $scope.neededCurrentPage = 1;
             });
         };
 
+        $scope.showRemoteNeed = function (device) {
+            $scope.remoteNeedDevice = device;
+            $scope.deviceFolders(device).forEach(function(folder) {
+                if ($scope.completion[device.deviceID][folder].needItems === 0) {
+                    return;
+                }
+                $scope.remoteNeedFolders.push(folder);
+                $scope.refreshRemoteNeed(folder, 1, 10);
+            });
+            $('#remoteNeed').modal().on('hidden.bs.modal', function () {
+                resetRemoteNeed()
+            });
```

Comment:
This will screw up if you close the dialog before the request returns. Not sure how to do it right. Move reset early?

Reply:
A solution might be to wrap [saving the return value from the request](https://github.com/syncthing/syncthing/pull/4559/files/131d3daa64ceae10263348eff8eb5a4b26f85c4d#diff-10f45c616bfdd371810aecdd203d0f36R673) inside a `if ($scope.remoteNeedDevice !== '')` thus no new data will be added after the modal is closed. That is probably racy though, so additionally resetting early to catch that is probably still a good idea. I don't like to exclusively reset early to not keep that data pointlessly in memory.

Reply:
As this isn't marked as "outdated" I addressed this by only adding data to the object if the modal is still open and resetting information at the beginning as well just for good measure.